index.js

```diff
@@ -1,29 +1,46 @@
 'use strict';
 var dgram = require('dgram');
-var onetime = require('onetime');
 var roots = require('root-hints')('A');
 var isReachable = require('is-reachable');
 var randomItem = require('random-item');
 var objectAssign = require('object-assign');
 var hostnames = require('./hostnames');
+var Promise = require('pinkie-promise');
 
 var timeout = 2000;
 
-module.exports = function (options, cb) {
-	if (typeof options === 'function') {
-		cb = options;
-		options = {};
-	}
 
-	options = objectAssign({ hostnames: hostnames, timeout: timeout }, options);
+module.exports = function (options) {
+	options = getOptions(options);
 
-	cb = onetime(cb);
+	var udpSocket = dgram.createSocket('udp4');
 
 	// Pick a random root server to query
 	var server = randomItem(roots);
 
-	// Craft a DNS query
-	var payload = new Buffer([
+	var prom = new Promise(function (resolve, reject) {
```

Comment:
Not much value in having multiple nested promises here. I think it would be better to just wrap the whole exported function in a promise.